### ISSUE TITLE
HMR broken in 1.x when changing publicPath

### ISSUE THREAD

**Issue by jbbr** [JeffreyWay/laravel-mix#923]
### Description:
HMR is failing after upgrading my Laravel Mix setup in a non-laravel project to 1.x.
Problem is that `setPublicPath` or `Config.publicPath = './something'` is not working everywhere.

- The `hot` file is created in the same directory as the `webpack.mix.js`
- The `hot` file is not getting deleted
- The full, absolute path is used in the `mix-manifest.json` instead of the path relative to the `publicPath`

I think the problem is, that on some places in the code the value of `Config.publicPath` is used before it can be changed using `setPublicPath`. e.g:
https://github.com/JeffreyWay/laravel-mix/blob/master/src/index.js#L48
`Config.publicPath` is empty in index.js and can't be modified from outside as the `Config` global is exported in this file.

Maybe the behavior is intended but its seems odd to me. Am I missing something?
Thanks

### Steps To Reproduce:

- Clone repo https://github.com/jbbr/laravel-mix-publicpath-bug-reproduce
- `$ npm run hot`

Problems:
- `hot` file is in `./hot` instead of `./htdocs/hot`
- paths in `mix-manifest.json` are prefixed with `/htdocs` (which should not be the case)

### Environment:
- Laravel Mix Version: 1.0.3
- Node Version: 8.1.2
- NPM Version: 4.6.1
- OS: Linux

**Comment by u01jmg3** [comment]
I too am experiencing this problem.

When I run `npm run hot`, npm fails saying that it cannot find `path/to/application/public/hot` even though in `webpack.mix.js` I have customised `publicPath` to be elsewhere. `npm run dev` and others complete without issue.

I think the problem is as you say:
> _... that on some places in the code the value of `Config.publicPath` is used before it can be changed using `setPublicPath`._

I also tried setting `devServer.contentBase` in case that would work but it doesn't.
